include/vcpkg/sourceparagraph.h

```diff
@@ -1,22 +1,26 @@
 #pragma once
 
-#include <vcpkg/base/fwd/json.h>
-
 #include <vcpkg/fwd/configuration.h>
 #include <vcpkg/fwd/vcpkgcmdarguments.h>
 
 #include <vcpkg/base/expected.h>
+#include <vcpkg/base/files.h>
+#include <vcpkg/base/json.h>
 #include <vcpkg/base/span.h>
-#include <vcpkg/base/system.h>
 
 #include <vcpkg/packagespec.h>
 #include <vcpkg/paragraphparser.h>
 #include <vcpkg/platform-expression.h>
-#include <vcpkg/versiondeserializers.h>
 #include <vcpkg/versions.h>
 
 namespace vcpkg
 {
+    struct ManifestAndPath
```

Comment:
Why did this get moved up here? I don't see it used in this file....

Reply:
`ManifestAndPath`'s definition requires intimate knowledge of JSON due to it being a data member. I therefore think it should be declared somewhere else than `vcpkgpaths.h` as part of "reduce the size of the world" -- lots of people calling vcpkgpaths.h don't need a manifest.

Looking at our files, `sourceparagraph.h` is the file that actually logically "describes" our manifest information (structure, fields, parsing, etc). This leads me to two thoughts:

1. I think anyone asking for `ManifestAndPath` will also need to include this file to get other information about the manifest structure
2. If the manifest structure is changing, this file will also change.

So I think moving `ManifestAndPath` here is a net neutral or overall reduces our "Concepts x Files" count.

Reply:
The effect of this change the opposite: it exposes the struct to more places, because `vcpkgpaths.h` includes `sourceparagraph.h`, so moving it here exposes it to strictly more code.

Crucially, this also forces `sourceparagraph.h` to include `files.h` to get `Path`, adding a real dependency where previously none existed.

Reply:
(I would be in support of a change to make it possible to talk to Paths without necessarily dragging in all of the manifest stuff but I think this movement should be part of that change, not this one)